src/App.tsx

```diff
@@ -47,37 +57,102 @@ export function getReorderedGoods(
 //   sortType: SortType,
 // };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
-
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
-
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean;
+  sortType: SortType;
+};
+
+const defaultState: State = {
+  isReversed: false,
+  sortType: SortType.NONE,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = defaultState;
+
+  reverseOrder = () => {
+    this.setState(state => {
+      return {
+        isReversed: !state.isReversed,
+      };
+    });
+  };
+
+  sortGoods = (sortType: SortType) => {
+    this.setState(() => {
+      return {
+        sortType: sortType,
+      };
+    });
+  };
+
+  sortReset = () => {
+    this.setState(() => {
+      return defaultState;
+    });
+  };
+
+  stateChanged = () => {
+    return (
+      this.state.isReversed !== defaultState.isReversed ||
+      this.state.sortType !== defaultState.sortType
+    );
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+
+    const reorderedGoods = getReorderedGoods(goodsFromServer, {
+      isReversed,
+      sortType,
+    });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            onClick={() => this.sortGoods(SortType.ALPHABET)}
+            className={`button is-info ${sortType === SortType.ALPHABET ? '' : 'is-light'}`}
```

Comment:
why empty classname  -''?
you can use this library just like 'classname': condition